pkg/v4/resource/chart/update.go

```diff
@@ -124,7 +127,7 @@ func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desir
 	r.logger.LogCtx(ctx, "level", "debug", "message", "finding out if chart-operator has to be updated")
 
 	updateState := &ResourceState{}
-	if !reflect.DeepEqual(currentResourceState, ResourceState{}) && !reflect.DeepEqual(currentResourceState, desiredResourceState) {
+	if currentResourceState.ReleaseVersion != "" && isChartModified(currentResourceState, desiredResourceState) {
```

Comment:
I mentioned an `Equals` method above. Maybe it should just be a `IsChartModified` method considering all the jazz here.

Reply:
Good point, I've added a `shouldUpdate` function with all the checks, in this case only for the `ReleaseVersion` field, PTAL.